src/playlist-loader.js

```diff
@@ -62,19 +64,29 @@ const addLLHLSQueryDirectives = (uri, media) => {
     }
 
     // add _HLS_msn= in front of any _HLS_part query
-    query.unshift(`_HLS_msn=${nextMSN}`);
+    // eslint-disable-next-line
+    parameters._HLS_msn = nextMSN;
   }
 
   if (media.serverControl && media.serverControl.canSkipUntil) {
     // add _HLS_skip= infront of all other queries.
-    query.unshift('_HLS_skip=' + (media.serverControl.canSkipDateranges ? 'v2' : 'YES'));
+    // eslint-disable-next-line
+    parameters._HLS_skip = (media.serverControl.canSkipDateranges ? 'v2' : 'YES');
   }
 
-  query.forEach(function(str, i) {
-    const symbol = i === 0 ? '?' : '&';
+  if (Object.keys(parameters).length) {
+    const parsedUri = new window.URL(uri);
 
-    uri += `${symbol}${str}`;
-  });
+    ['_HLS_skip', '_HLS_msn', '_HLS_part'].forEach(function(name) {
```

Comment:
I wanted to make sure that we keep the parameters in an expected order of `skip`, `msn`, and `part` as it makes reading the urls much easier.